test/functional/mempool_accept_wtxid.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test mempool acceptance in case of an already known transaction
+with identical non-witness data different witness.
+"""
+
+from test_framework.script import (
+    CScript,
+    OP_0,
+    OP_TRUE,
+    OP_IF,
+    OP_HASH160,
+    OP_EQUAL,
+    OP_ELSE,
+    OP_ENDIF,
+    hash160,
+)
+from test_framework.messages import (
+    CTransaction,
+    CTxIn,
+    CTxInWitness,
+    CTxOut,
+    COutPoint,
+    sha256,
+    COIN,
+    ToHex,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+
+class MempoolWtxidTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+        self.extra_args = [["-incrementalrelayfee=0"]]
+
+    def run_test(self):
+        node = self.nodes[0]
+
+        self.log.info('Start with empty mempool, and 200 blocks')
+        privkeys = [node.get_deterministic_priv_key().key]
+        coins = []
+        # The last 100 coinbase transactions are premature
+        for b in node.generate(110)[:-100]:
+            coinbase = node.getblock(blockhash=b, verbosity=2)["tx"][0]
+            coins.append({
+                "txid": coinbase["txid"],
+                "amount": coinbase["vout"][0]["value"],
+                "scriptPubKey": coinbase["vout"][0]["scriptPubKey"],
+            })
+        assert_equal(node.getmempoolinfo()['size'], 0)
+
+        txid = coins.pop()["txid"]
```

Comment:
You don't need a `coins` list or all this logic. Just generate 101 blocks, then get the txid of the first transaction in the tip block.